bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadPoolManager.java

```diff
@@ -0,0 +1,247 @@
+        return (cfg != null) ? cfg : new int[] { DEFAULT_THREAD_POOL_SIZE, DEFAULT_THREAD_POOL_CORE_SIZE };
+    }
+
+    private static class CommonThreadExecutor extends ThreadPoolExecutor {
+
+        public CommonThreadExecutor(final String poolName, int corePoolSize, int maxPoolSize, int maxQueueSize) {
+            this(poolName, corePoolSize, maxPoolSize, maxQueueSize, new NamedThreadFactory(poolName),
+                    new ThreadPoolExecutor.DiscardPolicy() {
+                        // The pool is bounded and rejections will happen during shutdown
+                        @Override
+                        public void rejectedExecution(Runnable runnable, ThreadPoolExecutor threadPoolExecutor) {
+                            // Log and discard
+                            logger.warn("Thread pool '{}' rejected execution of {}",
+                                    new Object[] { poolName, runnable.getClass() });
+                            super.rejectedExecution(runnable, threadPoolExecutor);
+                        }
+                    });
+        }
+
+        public CommonThreadExecutor(String threadPool, int corePoolSize, int maxPoolSize, int maxQueueSize,
+                ThreadFactory threadFactory, RejectedExecutionHandler rejectedHandler) {
+            // This is the same as Executors.newCachedThreadPool
+            super(corePoolSize, maxPoolSize, THREAD_TIMEOUT, TimeUnit.SECONDS,
+                    new ArrayBlockingQueue<Runnable>(maxQueueSize), threadFactory, rejectedHandler);
+        }
+
+        @Override
+        protected void afterExecute(Runnable runnable, Throwable throwable) {
+            super.afterExecute(runnable, throwable);
+            if (throwable != null) {
+                Throwable cause = throwable.getCause();
+                if (cause instanceof InterruptedException) {
+                    // Ignore this, might happen when we shutdownNow() the executor. We can't
+                    // log at this point as the logging system might be stopped already.
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * This is a normal thread factory, which adds a named prefix to all created threads.
+     */
+    private static class NamedThreadFactory implements ThreadFactory {
+
+        protected final ThreadGroup group;
+        protected final AtomicInteger threadNumber = new AtomicInteger(1);
+        protected final String namePrefix;
+
+        public NamedThreadFactory(String threadPool) {
+            this.namePrefix = "ESH-" + threadPool + "-";
+            SecurityManager s = System.getSecurityManager();
+            group = (s != null) ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
+        }
+
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = new Thread(group, r, namePrefix + threadNumber.getAndIncrement(), 0);
+            if (t.isDaemon())
+                t.setDaemon(false);
```

Comment:
Why not use Daemon threads? at least they prevent the JVM from blocking at the exit. if the thread pool manager is used by other parts of the core or bindings, and if the bindings are a bit dodgy, at least we know that it won't hurt the JVM... no?